### DISCUSSION TITLE
Creating a Release configuration manifest

### DISCUSSION THREAD

**Discussion by peteraritchie** [microsoft/aspire#1189] [Q&A] [answered]
Having looked at `AddNodeApp` (initially via [How to include an Angular Project within .NET Aspire](https://timdeschryver.dev/blog/how-to-include-an-angular-project-within-net-aspire)), I'm wondering how to generate a Release .NET Aspire manifest.  Using `AddNodeApp` and [building a manifest](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/manifest-format#code-try-1), I can get a manifest that contains the npm resource, but the `env/NODE_ENV` is always "development.".  I've tried setting the configuration for MSBuild:

`dotnet run -configuration Release --project AspireSample.AppHost\AspireSample.AppHost.csproj -- --configuration Release --publisher manifest --output-path aspire-manifest.json`

...and for the app:

`dotnet run --project AspireSample.AppHost\AspireSample.AppHost.csproj -- --publisher manifest --configuration Release  --output-path aspire-manifest.json`

But both produce a manifest containing "development".

What am I missing?

**Comment by DamianEdwards** [comment]
Try generating the manifest using the MSBuild target rather than `dotnet run` perhaps? e.g. `dotnet msbuild -t:GenerateAspireManifest`

**Reply by peteraritchie** [reply]
Hmm, the default (with no other options, which I expect to default to Debug), results in a production manifest output to the `Debug` output directory.  If I include configuration environment e.g., ` dotnet msbuild -t:GenerateAspireManifest -property:Configuration=Debug` it still generates a production manifest to the Debug output directory.  A Release configuration does output a release manifest to the `Release` output directory (e.g., ` dotnet msbuild -t:GenerateAspireManifest -property:Configuration=Release`.)

**Reply by DamianEdwards** [reply, marked as answer]
The concept of build configurations and environments (development, production) are independent and not tied together directly. The `dotnet run` command will select the first launch profile from the *launchSettings.json* and honor whatever environment variables are configured in there, which by default includes setting the environment to "Development". `dotnet run` also defaults to using the Debug build configuration. `dotnet msbuild` doesn't use launch profiles so the default environment of "Production" will be used.

**Reply by peteraritchie** [reply]
Very helpful, thanks @DamianEdwards !
